dev/tools/gen_date_localizations.dart

```diff
@@ -0,0 +1,147 @@
+    );
+  }
+
+  final File dotPackagesFile = new File(path.join('packages', 'flutter', '.packages'));
+  final bool dotPackagesExists = dotPackagesFile.existsSync();
+
+  if (!dotPackagesExists) {
+    _fatal(
+      'File not found: ${dotPackagesFile.path}. $_kCommandName must be run '
+      'after a successful "flutter update-packages".'
+    );
+  }
+
+  final String pathToIntl = dotPackagesFile
+    .readAsStringSync()
+    .split('\n')
+    .firstWhere(
+      (String line) => line.startsWith('intl:'),
+      orElse: () {
+        _fatal('intl dependency not found in ${dotPackagesFile.path}');
+      },
+    )
+    .split(':')
+    .last;
+
+  final Directory dateSymbolsDirectory = new Directory(path.join(pathToIntl, 'src', 'data', 'dates', 'symbols'));
+  final Map<String, File> symbolFiles = _listIntlData(dateSymbolsDirectory);
+  final Directory datePatternsDirectory = new Directory(path.join(pathToIntl, 'src', 'data', 'dates', 'patterns'));
+  final Map<String, File> patternFiles = _listIntlData(datePatternsDirectory);
+  final List<String> materialLocales = _materialLocales().toList();
+  final StringBuffer buffer = new StringBuffer();
+
+  buffer.writeln(
+'''
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file has been automatically generated.  Please do not edit it manually.
+// To regenerate run (omit -w to print to console instead of the file):
+// dart dev/tools/gen_date_localizations.dart -w
+
+'''
+);
+  buffer.writeln('const Map<String, dynamic> dateSymbols = const <String, dynamic> {');
+  symbolFiles.forEach((String locale, File data) {
+    if (materialLocales.contains(locale))
+      buffer.writeln(_jsonToMapEntry(locale, JSON.decode(data.readAsStringSync())));
+  });
+  buffer.writeln('};');
+
+  buffer.writeln('const Map<String, dynamic> datePatterns = const <String, dynamic> {');
+  patternFiles.forEach((String locale, File data) {
+    if (materialLocales.contains(locale))
+      buffer.writeln(_jsonToMapEntry(locale, JSON.decode(data.readAsStringSync())));
+  });
+  buffer.writeln('};');
+
+  if (writeToFile) {
+    final File dateLocalizationsFile = new File(path.join('packages', 'flutter', 'lib', 'src', 'material', 'i18n', 'date_localizations.dart'));
+    dateLocalizationsFile.writeAsStringSync(buffer.toString());
+    Process.runSync(path.join('bin', 'cache', 'dart-sdk', 'bin', 'dartfmt'), <String>[
+      '-w',
+      dateLocalizationsFile.path,
+    ]);
+  } else {
+    print(buffer);
+  }
+}
+
+String _jsonToMapEntry(String key, dynamic value) {
+  return "'$key': ${_jsonToMap(value)},";
+}
+
+String _jsonToMap(dynamic json) {
+  if (json == null || json is num || json is bool)
+    return '$json';
+
+  if (json is String) {
+    if (json.contains("'"))
+      return 'r"""$json"""';
+    else
+      return "r'''$json'''";
+  }
+
+  if (json is Iterable)
```

Comment:
You could generate const <String> or const <int> here with just a little more code.